pipeline/cactus_workflow.py

```diff
@@ -582,82 +661,83 @@ def runBarForTarget(self, calculateWhichEndsToComputeSeparately=None, endAlignme
+                    alignmentID = self.addChild(CactusBarEndAlignerWrapper(self.phaseNode, self.constantsNode, self.cactusDiskDatabaseString, self.flowerNames, False, endsToAlign)).rv()
+                    precomputedAlignmentIDs.append(alignmentID)
                     endsToAlign = []
                     totalSize = 0
-                    alignmentFileCount += 1
         if len(endsToAlign) > 0:
-            self.addChildTarget(CactusBarEndAlignerWrapper(self.phaseNode, self.constantsNode, self.cactusDiskDatabaseString, self.flowerNames, 
-                                                           False, endsToAlign, os.path.join(self.getGlobalTempDir(), "endAlignments.%i" % alignmentFileCount)))
-            alignmentFileCount += 1
-        self.phaseNode.attrib["precomputedAlignmentFiles"] = " ".join([ os.path.join(self.getGlobalTempDir(), ("endAlignments.%i") % i) for i in range(alignmentFileCount) ]) 
-        self.makeFollowOnRecursiveTarget(CactusBarWrapperWithPrecomputedEndAlignments)
-        self.logToMaster("Breaking bar job into %i separate jobs" % \
-                             (alignmentFileCount))
+            precomputedAlignmentIDs.append(self.addChild(CactusBarEndAlignerWrapper(self.phaseNode, self.constantsNode, self.cactusDiskDatabaseString, self.flowerNames, False, endsToAlign)).rv())
+        self.precomputedAlignmentIDs = precomputedAlignmentIDs
+        self.makeFollowOnRecursiveJob(CactusBarWrapperWithPrecomputedEndAlignments)
+        fileStore.logToMaster("Breaking bar job into %i separate jobs" % \
+                             (len(precomputedAlignmentIDs)))
         
-class CactusBarEndAlignerWrapper(CactusRecursionTarget):
+class CactusBarEndAlignerWrapper(CactusRecursionJob):
     """Computes an end alignment.
     """
-    def __init__(self, phaseNode, constantsNode, cactusDiskDatabaseString, flowerNames, overlarge, endsToAlign, alignmentFile):
-        CactusRecursionTarget.__init__(self, phaseNode, constantsNode, cactusDiskDatabaseString, flowerNames, overlarge)
+    def __init__(self, phaseNode, constantsNode, cactusDiskDatabaseString, flowerNames, overlarge, endsToAlign):
+        CactusRecursionJob.__init__(self, phaseNode, constantsNode, cactusDiskDatabaseString, flowerNames, overlarge)
         self.endsToAlign = endsToAlign
-        self.alignmentFile = alignmentFile
     
-    def run(self):
+    def run(self, fileStore):
         self.endsToAlign = [ int(i) for i in self.endsToAlign ]
         self.endsToAlign.sort()
         self.flowerNames = encodeFlowerNames((decodeFirstFlowerName(self.flowerNames),) + tuple(self.endsToAlign)) #The ends to align become like extra flower names
-        messages = runBarForTarget(self, 
-                                   endAlignmentsToPrecomputeOutputFile=self.alignmentFile)
+        alignmentFile = fileStore.getLocalTempFile()
```

Comment:
This will break, but only on whole-genome runs. No rush, as there are probably other things to do first.

If it would help, I can point you at some mid-size genomes that will take hours, rather than days, to run, but will still trigger these annoying paths that don't get run on small alignments.